README.md

```diff
@@ -1,4 +1,127 @@
-Universal Transfer Operator
+<h1 align="center">
+  Universal Transfer Operator
+</h1>
+  <h3 align="center">
+transfers made easy<br><br>
+</h3>
 
-To build locally:
-```make container target=build-run```
+
+
+[![CI](https://github.com/astronomer/apache-airflow-provider-transfers/actions/workflows/ci-uto.yaml/badge.svg)](https://github.com/astronomer/apache-airflow-provider-transfers)
+
+The **Universal Transfer Operator** allows data transfers between any supported source and target Datasets in [Apache Airflow](https://airflow.apache.org/). It offers a consistent agnostic interface, simplifying the users' experience, so they do not need to use specific providers or operators for transfers. The Astro Python SDK is maintained by [Astronomer](https://astronomer.io).
+
+This ensures a consistent set of data providers that can read from and write to dataset. The Universal Transfer
+Operator can use the respective data providers to transfer between as a source and a destination. It also takes advantage of any existing fast and
+direct high-speed endpoints, such as Snowflake's built-in ``COPY INTO`` command to load S3 files efficiently into Snowflake.
+
+Universal transfer operator also supports the transfers using third-party platforms like Fivetran.
+
+## How Universal Transfer Operator Works
+
+
+![Approach](./docs/images/approach.png)
+
+With universal transfer operator, users can perform data transfers using the following transfer modes:
+
+1. non-native
+2. native
+3. third-party
+
+### Non-native transfer
+
+When we load a data located in one dataset located in cloud to another dataset located in cloud, internally the steps involved are:
+
+Steps:
+
+- Get the dataset data in chunks from dataset storage to the worker node.
+- Send data to the cloud dataset from the worker node.
+
+This is the default way of transferring datasets. There are performance bottlenecks because of limitations of memory, processing power, and internet bandwidth of worker node.
+
+Following is an example of non-native transfers between Google cloud storage and Sqlite:
+
+https://github.com/astronomer/apache-airflow-provider-transfers/blob/main/example_dags/example_universal_transfer_operator.py#L37-L41
+
+### Improving bottlenecks by using native transfer
+
+Some datasets on cloud like Bigquery and Snowflake support native transfer to ingest data from cloud storage directly. Using this we can ingest data much quicker and without any involvement of the worker node.
+
+Steps:
+
+- Request destination dataset to ingest data from the file dataset.
+- Destination dataset request source dataset for data.
+
+This is a faster way for datasets of larger size as there is only one network call involved and usually the bandwidth between vendors is high. Also, there is no requirement for memory/processing power of the worker node, since data never gets on the node. There is significant performance improvement due to native transfers.
```

Comment:
```suggestion
This is a faster way to transfer datasets of larger size as there is only one network call involved and usually the bandwidth between vendors is high. Also, there is no requirement for memory/processing power of the worker node, since data never gets on the node. There is significant performance improvement due to native transfers.
```